index.js

```diff
@@ -52,7 +52,7 @@ function ensureAllCommentsPrinted(astComments) {
 function format(text, opts, addAlignmentSize) {
   addAlignmentSize = addAlignmentSize || 0;
 
-  const ast = parser.parse(text, opts);
+  const ast = opts.astTransform(parser.parse(text, opts));
```

Comment:
I love the idea of supporting alternative parsers, but I'm not sure this is the place to do it, for a couple of reasons:

* It overloads "transform" to mean "parse and transform", and there's already a `parser` option, so I think this would be a somewhat confusing API.
* It pushes the task of parsing out into the transform function. At first, this sounds like I'm repeating the previous point, but the current design allows for independent AST transformation modules to be published and then used from the CLI, like this: `prettier --transform node_modules/some-npm-package-that-fixes-your-ast`. A couple of notes on this syntax:
  * We can get rid of "node_modules/" using something like [`resolve`](https://www.npmjs.com/package/resolve)
  * These could even be chained in the future e.g. `prettier --transform ast-sort-keys --transform ast-sort-objects`

  If we change the API to require the transformation function to also do the parsing, we lose that pluggability.

What if we instead let the `parser` option optionally take a function instead of a string? That would allow alternative parsers to be used, in a more intuitive/modular way (IMO).